addons/stock/models/stock_quant.py

```diff
@@ -412,7 +412,7 @@ def _update_available_quantity(self, product_id, location_id, quantity, lot_id=N
 
         for quant in quants:
             try:
-                with self._cr.savepoint():
+                with self._cr.savepoint(flush=False):  # Avoid flush compute store of package
```

Comment:
Maybe it would be worth doing the flush after the loop then?

Reply:
If I do, it will be useless for the scenario of tracked by serial number (case of the client), because there will be X call of `_update_available_quantity` and `quants` contains only one record in that case. Then, It doesn't solve anything.

Also I don't see any reason to flush anything here, better to let the ORM do is job, no ? I don't find any counter-example in this specific case where the flush will be useful.

Reply:
Well to be honest I don't know. But I guess in this case, the `savepoint` is only useful in case of PG error, so indeed there is no reason to flush beforehand.